SubmittingPatches.rst

```diff
@@ -62,468 +66,207 @@ then you just add a line saying ::
-title. Ideally it is a single short line of at most 72 characters,
-summarizing the change. It is required to prefix it with the
-subsystem or module you are changing. For instance, the prefix
-could be "doc:", "osd:", or "common:". One can use::
+So a typical commit message for revising the document could look like::
 
-     git log
+     doc: add "--foo" option to bar
 
-for more examples. It is also conventional to use the imperative mood in the
-commit title. For example::
+     * update the man page for bar with the newly added "--foo" option.
+     * fix a typo
 
-     mds: add perf counter for finisher of MDSRank
+     Fixes: http://tracker.ceph.com/issues/12345
+     Signed-off-by: Random J Developer <random@developer.example.org>
 
-or::
+If a commit fixes a regression introduced by a different commit, please
+add a line referencing the SHA1 of the commit that introduced the regression.
+For example::
 
-     osd: make the ClassHandler::mutex private
+     Fixes: 9dbe7a003989f8bb45fe14aaa587e9d60a392727
 
-For GitHub, please use this "subsystem: short description" convention for
-naming pull requests (PRs). These titles feed directly into the script that
-generates release notes and it is tedious to clean up non-conformant PR titles
-at release time.  This document places no limit on the length of PR titles, but
-be aware that they are subject to editing as part of the release process.
 
-Commit message
-^^^^^^^^^^^^^^
+.. _`commit message`:
 
-Be as specific as possible. The WORST descriptions possible include
-things like "update driver X", "bug fix for driver X", or "this patch
-includes updates for subsystem X. Please apply."
+Writing commit messages
+-----------------------
 
-If your description starts to get long, that's a sign that you probably
-need to split up your patch. See :ref:`split_changes`.
+(This section is about the body of the commit message. Please also see
+the `Title of commit`_ section for advice on titling commit messages.)
 
-When you submit or resubmit a patch or patch series, include the
-complete patch description and justification for it. Don't just
-say that this is version N of the patch (series). Don't expect the
-patch merger to refer back to earlier patch versions or referenced
-URLs to find the patch description and put that into the patch.
-I.e., the patch (series) and its description should be self-contained.
-This benefits both the patch merger(s) and reviewers. Some reviewers
-probably didn't even receive earlier versions of the patch.
+In the body of your commit message, be as specific as possible. The WORST
+descriptions possible include things like "update component X", "bug fix for
+component X", or "this patch includes updates for subsystem X.  Please apply."
 
-Tag the commit
-^^^^^^^^^^^^^^
+There's nothing inherently wrong with a long commit message body. But it might
+be a sign that the patch could benefit from being split up into smaller patches.
```

Comment:
I think we want to encourage long explanatory commit messages and this paragraph seems to discourage it. A long commit message may include error output (from teuthology) or other useful information to give background for the patch.

Let's instead emphasize, again, that a commit and its message should be focused around a particular change and if they have a series of bulleted modifications, those would be better split into multiple commits.

Reply:
i love explanatory commit message, and i love short commit messages as long as they are well written and they explain the whole story.

sometimes, i just don't have time to make it short.